File: IvmpDotNetWrapper/VehicleManager.cpp
#include "SDK\SDK.h"

EXPORT EntityId GetVehicleManager_Add(int iModelId, CVector3 vecSpawnPosition, CVector3 vecSpawnRotation, BYTE byteColor1, BYTE byteColor2, BYTE byteColor3, BYTE byteColor4, int respawn_delay = 1) {
	return GetVehicleManager()->Add(iModelId, vecSpawnPosition, vecSpawnRotation, byteColor1, byteColor2, byteColor3, byteColor4, 1);
}

EXPORT void GetVehicleManager_Remove(EntityId vehicleId) {
	GetVehicleManager()->Remove(vehicleId);
}

EXPORT bool GetVehicleManager_DoesExist(EntityId vehicleId) {
	return GetVehicleManager()->DoesExist(vehicleId);
}

EXPORT CVehicleInterface * GetVehicleManager_GetAt(EntityId vehicleId) {
	return GetVehicleManager()->GetAt(vehicleId);
}

File: IvmpDotNet.Core/Wrappings/VehicleManager.cs
﻿using IvmpDotNet.SDK;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IvmpDotNet.Wrappings {
    public class VehicleManager : IVehicleManager {
        public IVehicle Create(VehicleModels iModelId, CVector3 vecPosition, CVector3 vecRotation, int color1, int color2, int color3, int color4, int respawn_delay) {
            ushort id = (ushort)Imports.Vehicles.Vehicles_Create((int)iModelId, vecPosition, vecRotation, color1, color2, color3, color4, respawn_delay);
            return new Vehicle() {
                VehicleId = id
            };
        }

        public void Remove(ushort vehicleId) {
            Imports.GetVehicleManager.GetVehicleManager_Remove(vehicleId);
        }

        public bool DoesExist(ushort vehicleId) {
            return DoesExist(vehicleId);
        }
    }
}
